mantidproject/mantid
# Subtract background from all runs on loading in EngDiff UI

**Description of work.**
Subtract background automatically from all runs on loading - requested behaviour from instrument scientist. It's not possible to just change default value of bgsub checkbox to be True because the various dicts in model haven't been created. Instead, the bgsub checkboxes of runs just loaded are checked to be True after the table ahs first been populated (behaves same as if the user has gone through and checked them all by hand).  To ensure the table is repopulated with the correct bgsub status I have added a setter to the model.

**To test:**
(1) In the main workbench settings, set the peak function to a `Gaussian` (File > Settings >Fitting > Default peak shape)
(2) Open the EngDiff UI (Interfaces > Diffraction > Engineering Diffraction)
(3) Go to the fitting tab, load in some focussed data - you will need two runs (to make focussed data follow the instructions in Test 1 here https://developer.mantidproject.org/Testing/EngineeringDiffraction/EngineeringDiffractionTestGuide.html or get the data from PR #29726)
(4) Load in one run - note the bgsub checkbox should be checked
(5) Uncheck the bgsub checkbox
(6) Load in another run - note the bg has been subtracte dform new run but the checkbox status of old run is not overwritten.



Fixes #31689 





---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
